### PR TITLE
Add Gzipping CloudWatch client

### PR DESCRIPTION
**Pull request by john-eriksson-imperva**
In the previous Rusoto-based version of metrics_cloudwatch, it was possible to enable GZIP compression of the data in PutMetricData by passing a Rusoto client created with `rusoto_core::Client::new_with_encoding` to metrics_cloudwatch. 

This PR creates a new `GzipClient` that implements the `Cloudwatch` trait and adds GZIP compression to the request.

One thing that could be discussed is how to improve the developer experience. You would have to use the client method in the builder to use this GzipClient. There are problems with just adding a gzip method to the builder, as GzipClient can only wrap a regular `aws_sdk_cloudwatch::Client`.

### CODE REVIEW

**Comment by VladimirBramstedt** [comment]
A thought: We can make this cleaner by making the Client trait private. it was only really ever used for mocking and tests.
if we can simplify and make the builder behave more naturally, i'd be in favour of doing a 3.0 release with a cleaned up interface. @ramn thoughts?

**Comment by ramn** [comment]
@VladimirBramstedt sure can try! And dogfood it!

**Comment by john-eriksson-imperva** [comment]
I'm not so sure about the mocking and the usages of `dyn Cloudwatch`. We only have two implementations yet of `Cloudwatch`, the real implementation and a mock implementation that is only used internally. If we get rid of the mock one, we will get rid of some virtual dispatch and improve the performance a bit. But that requires some other way to verify that we sent the right requests to Cloudwatch in the tests.

File: src/collector.rs [hunk 1]
```diff
@@ -26,29 +41,88 @@ use {
...
+    gzip_setting: GzipSetting,
+}
+
+#[cfg(feature = "gzip")]
+#[derive(Debug)]
+struct GzipInterceptor {
+    compression: flate2::Compression,
+}
+#[cfg(feature = "gzip")]
+impl Intercept for GzipInterceptor {
```
Comment (VladimirBramstedt) [inline src/collector.rs]: do we really need the Interceptor dance, or would it be enough using `customize().mutate_request(...)` ?

Comment (john-eriksson-imperva) [reply src/collector.rs]: You can use `mutate_request` to change it as well. But the `mutate_request` call creates an interceptor `aws_smithy_runtime::client::interceptors::MutateRequestInterceptor` with the passed closure and adds that one to the request, so in the end the AWS SDK will use interceptors anyway.

Comment (VladimirBramstedt) [reply src/collector.rs]: i know, but then you dont need to implement this trait or use it directly (the design of the sdk is awful, the fewer explicit things from there we use the better...). mostly wonder if it results in slightly less aws boilerplate to use mutate_request vs this approach?

**Pull request closed without merging**